### PR TITLE
Heatmap: new panel based based on uPlot

### PR DESCRIPTION
**Pull request by ryantxu**
new Heatmap panel based on uPlot rather than d3+flot

New: :mouse2: :racing_car:  (10x faster, 10x less memory)

image

Old: :turtle: :sloth: :elephant: 

image

### CODE REVIEW

File: packages/grafana-ui/src/components/VizTooltip/VizTooltipContainer.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ export const VizTooltipContainer: React.FC<VizTooltipContainerProps> = ({
   position: { x: positionX, y: positionY },
   offset: { x: offsetX, y: offsetY },
   children,
-  allowPointerEvents,
+  allowPointerEvents = false,
```
Comment (ryantxu) [inline packages/grafana-ui/src/components/VizTooltip/VizTooltipContainer.tsx:26]: is this something that should be merged and backported independently?

File: public/app/plugins/panel/xychart/scatter.ts [hunk 2]
```diff
@@ -328,11 +328,8 @@ const prepConfig = (
 
           let deg360 = 2 * Math.PI;
 
-          // leon forgot to add these to the uPlot's Scale interface, but they exist!
-          //let xKey = scaleX.key as string;
-          //let yKey = scaleY.key as string;
-          let xKey = series.facets![0].scale;
-          let yKey = series.facets![1].scale;
+          let xKey = scaleX.key!;
```
Comment (ryantxu) [inline public/app/plugins/panel/xychart/scatter.ts:331]: similarly... should this get merged independently?

File: packages/grafana-ui/src/components/VizTooltip/VizTooltipContainer.tsx (hunk 1 above)
Comment (ryantxu) [reply packages/grafana-ui/src/components/VizTooltip/VizTooltipContainer.tsx:26]: moved to https://github.com/grafana/grafana/pull/45492

File: public/app/plugins/panel/xychart/scatter.ts (hunk 2 above)
Comment (ryantxu) [reply public/app/plugins/panel/xychart/scatter.ts:331]: moved to https://github.com/grafana/grafana/pull/45492

**Review by leeoniya — APPROVED** [review]
this looks okay

**Pull request merged**